include/deal.II/base/type_traits.h

```diff
@@ -0,0 +1,126 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__type_traits_h
+#define dealii__type_traits_h
+
+
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/block_vector.h>
+#include <deal.II/lac/la_vector.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace std_cxx11
+{
+
+  /**
+  * A true_type similar to c++11 std::true_type
+  *
+  * @author Uwe Köcher, 2017
+  */
+  struct true_type
+  {
+    static const bool value = true;
+  };
+
+
+  /**
+  * A false_type similar to c++11 std::false_type
+  *
+  * @author Uwe Köcher, 2017
+  */
+  struct false_type
+  {
+    static const bool value = false;
+  };
+
+}
+
+
+
+/**
+ * Type trait for a serial vector, i.e. a vector class which is not distributed
+ * over processes.
+ *
+ * @author Uwe Köcher, 2017
+ */
+template<typename T>
+struct is_serial_vector : std_cxx11::false_type {};
+
+template<>
+struct is_serial_vector< Vector< float > > : std_cxx11::true_type
```

Comment:
You can (and probably should) actually write this as a partial specialization of the general template:
```
  template <typename Number>
  struct is_serial_vector<Vector<Number> > : std_cxx11::true_type {};
```
This covers *all* template arguments, and I think that's correct because the property you're describing here really is one of the vector class, not the (scalar type) template argument.